### ISSUE TITLE
Expection: name claim is missing 

### ISSUE THREAD

**Issue by lasrol** [IdentityServer/IdentityServer4#1465]
- [x] I read and understood how to enable [logging](https://identityserver4.readthedocs.io/en/release/topics/logging.html)

### Issue / Steps to reproduce the problem

I have an external identity provider that i allow my users to log in with. They return me a jwt with some claims, but not the name claim. So when you validate the principal and assert it has the required claims it throws an exception

`if (principal.FindFirst(JwtClaimTypes.Name) == null) throw new InvalidOperationException("name claim is missing");`

So i am curious why this is required? I can hook into the event OnTicketReceived and recreate the Identity Principal with a "fake" name claim, but that defeats the purpose and does not feel right. 

So basically my code never reaches ExternalCallback controller action that should handle and setup the user. Is it some configuration issue on my part, with the Authentication scheme?

I am using core 2, and your latest preview package.

**Comment by lasrol** [comment]
It seems that exception went away when changing the SignInAuthenticationScheme. Using idsrv.external as SignInAuthenticationScheme fixed the validation error, but the signInManager is not giving me any external info. 

I find the authentication and cookie configuration is somewhat confusing(Not your implementation, but from the MS team), but i guess i just don't understand it well enough. 

In any case, i am closing this issue as i don't think its related to the identity server.
